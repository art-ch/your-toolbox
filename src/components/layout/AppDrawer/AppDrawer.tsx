'use client';

import React from 'react';

import { Overlay } from './components/Overlay/Overlay';
import { Header } from './components/Header/Header';
import { Content } from './components/Content/Content';
import { Container } from './components/Container/Container';
import { Footer } from './components/Footer/Footer';

export const AppDrawer = () => {
  return (
    <div>
      <Overlay />
      <Container>
        <Header />
        <Content />
        <Footer />
      </Container>
    </div>
  );
};
