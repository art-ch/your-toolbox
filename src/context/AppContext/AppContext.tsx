import { createContext, useContext } from 'react';
import { DrawerState, useDrawer } from './hooks/useDrawer/useDrawer';

type AppContextType = {
  drawer: DrawerState;
};

const AppContext = createContext<AppContextType | null>(null);

export const AppProvider = ({ children }: { children: React.ReactNode }) => {
  const drawer = useDrawer();

  return (
    <AppContext.Provider value={{ drawer }}>{children}</AppContext.Provider>
  );
};

export const useAppContext = () => {
  const context = useContext(AppContext);

  if (!context) {
    throw new Error('useAppContext must be used within an AppProvider');
  }

  return context;
};
