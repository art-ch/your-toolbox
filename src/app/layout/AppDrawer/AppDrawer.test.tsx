import { render } from '@testing-library/react';

import { AppDrawer } from './AppDrawer';
import { AppProvider } from '@/app/context/AppContext/AppContext';

jest.mock('@/hooks/use-mobile', () => ({
  useIsMobile: jest.fn()
}));

describe('AppDrawer component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <AppProvider>
        <AppDrawer />
      </AppProvider>
    );

    expect(container).toMatchSnapshot();
  });
});
