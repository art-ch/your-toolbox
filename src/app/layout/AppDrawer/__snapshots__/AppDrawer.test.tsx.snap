// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDrawer component should render correctly 1`] = `
<div>
  <div>
    <div
      class="fixed top-0 left-0 z-20 bottom-auto transition-transform duration-300 ease-in-out shadow bg-white w-60 h-[100dvh] -translate-x-full"
      data-testid="container"
    >
      <div
        class="flex justify-between min-w-full p-2.5 shadow-[1px_1px_1px_0_rgba(0,0,0,0.06)]"
      >
        <a
          href="/navigation"
        >
          <h3
            class="font-semibold"
          >
            Navigation
          </h3>
        </a>
        <svg
          class="lucide lucide-x cursor-pointer"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 6 6 18"
          />
          <path
            d="m6 6 12 12"
          />
        </svg>
      </div>
      <div
        class="group/sidebar-wrapper flex w-full has-[[data-variant=inset]]:bg-sidebar min-h-0"
        style="--sidebar-width: 16rem; --sidebar-width-icon: 3rem;"
      >
        <div
          class="flex min-h-0 flex-1 flex-col gap-2 overflow-auto group-data-[collapsible=icon]:overflow-hidden"
          data-sidebar="content"
        >
          <div
            class="relative flex w-full min-w-0 flex-col p-2"
            data-sidebar="group"
          >
            <ul
              class="flex w-full min-w-0 flex-col gap-1"
              data-sidebar="menu"
            >
              <li
                class="group/menu-item relative"
                data-sidebar="menu-item"
              >
                <a
                  class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-none ring-sidebar-ring transition-[width,height,padding] focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-[[data-sidebar=menu-action]]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:!size-8 group-data-[collapsible=icon]:!p-2 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm font-medium"
                  data-active="false"
                  data-sidebar="menu-button"
                  data-size="default"
                  href="/calculators"
                >
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <rect
                      height="20"
                      rx="2"
                      width="16"
                      x="4"
                      y="2"
                    />
                    <line
                      x1="8"
                      x2="16"
                      y1="6"
                      y2="6"
                    />
                    <line
                      x1="16"
                      x2="16"
                      y1="14"
                      y2="18"
                    />
                    <path
                      d="M16 10h.01"
                    />
                    <path
                      d="M12 10h.01"
                    />
                    <path
                      d="M8 10h.01"
                    />
                    <path
                      d="M12 14h.01"
                    />
                    <path
                      d="M8 14h.01"
                    />
                    <path
                      d="M12 18h.01"
                    />
                    <path
                      d="M8 18h.01"
                    />
                  </svg>
                  Calculators
                </a>
                <ul
                  class="mx-3.5 flex min-w-0 translate-x-px flex-col gap-1 border-sidebar-border py-0.5 group-data-[collapsible=icon]:hidden ml-0 border-l-0 px-1.5"
                  data-sidebar="menu-sub"
                >
                  <li>
                    <a
                      class="flex h-7 min-w-0 -translate-x-px items-center gap-2 overflow-hidden rounded-md px-2 text-sidebar-foreground outline-none ring-sidebar-ring hover:bg-sidebar-accent hover:text-sidebar-accent-foreground focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 aria-disabled:pointer-events-none aria-disabled:opacity-50 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 [&>svg]:text-sidebar-accent-foreground data-[active=true]:bg-sidebar-accent data-[active=true]:text-sidebar-accent-foreground text-sm group-data-[collapsible=icon]:hidden"
                      data-sidebar="menu-sub-button"
                      data-size="md"
                      href="/calculators/ascetic-exercise-duration"
                    >
                      Ascetic Exercise Duration
                    </a>
                  </li>
                </ul>
              </li>
              <li
                class="group/menu-item relative"
                data-sidebar="menu-item"
              >
                <a
                  class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-none ring-sidebar-ring transition-[width,height,padding] focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-[[data-sidebar=menu-action]]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:!size-8 group-data-[collapsible=icon]:!p-2 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm font-medium"
                  data-active="false"
                  data-sidebar="menu-button"
                  data-size="default"
                  href="/specialized-tools"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="m14.17 13.71 1.4-2.42c.09-.15.05-.34-.08-.45l-1.48-1.16c.03-.22.05-.45.05-.68s-.02-.46-.05-.69l1.48-1.16c.13-.11.17-.3.08-.45l-1.4-2.42c-.09-.15-.27-.21-.43-.15l-1.74.7c-.36-.28-.75-.51-1.18-.69l-.26-1.85a.364.364 0 0 0-.35-.29h-2.8c-.17 0-.32.13-.35.3L6.8 4.15c-.42.18-.82.41-1.18.69l-1.74-.7c-.16-.06-.34 0-.43.15l-1.4 2.42c-.09.15-.05.34.08.45l1.48 1.16c-.03.22-.05.45-.05.68s.02.46.05.69l-1.48 1.16c-.13.11-.17.3-.08.45l1.4 2.42c.09.15.27.21.43.15l1.74-.7c.36.28.75.51 1.18.69l.26 1.85c.03.16.18.29.35.29h2.8c.17 0 .32-.13.35-.3l.26-1.85c.42-.18.82-.41 1.18-.69l1.74.7c.16.06.34 0 .43-.15zM8.81 11c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zM21.92 18.67l-.96-.74c.02-.14.04-.29.04-.44 0-.15-.01-.3-.04-.44l.95-.74c.08-.07.11-.19.05-.29l-.9-1.55c-.05-.1-.17-.13-.28-.1l-1.11.45c-.23-.18-.48-.33-.76-.44l-.17-1.18a.216.216 0 0 0-.21-.2h-1.79c-.11 0-.21.08-.22.19l-.17 1.18c-.27.12-.53.26-.76.44l-1.11-.45a.23.23 0 0 0-.28.1l-.9 1.55c-.05.1-.04.22.05.29l.95.74a3.145 3.145 0 0 0 0 .88l-.95.74c-.08.07-.11.19-.05.29l.9 1.55c.05.1.17.13.28.1l1.11-.45c.23.18.48.33.76.44l.17 1.18c.02.11.11.19.22.19h1.79c.11 0 .21-.08.22-.19l.17-1.18c.27-.12.53-.26.75-.44l1.12.45c.1.04.22 0 .28-.1l.9-1.55c.06-.09.03-.21-.05-.28zm-4.29.16a1.35 1.35 0 1 1 .001-2.701 1.35 1.35 0 0 1-.001 2.701z"
                    />
                  </svg>
                  Specialized Tools
                </a>
                <ul
                  class="mx-3.5 flex min-w-0 translate-x-px flex-col gap-1 border-sidebar-border py-0.5 group-data-[collapsible=icon]:hidden ml-0 border-l-0 px-1.5"
                  data-sidebar="menu-sub"
                >
                  <li>
                    <a
                      class="flex h-7 min-w-0 -translate-x-px items-center gap-2 overflow-hidden rounded-md px-2 text-sidebar-foreground outline-none ring-sidebar-ring hover:bg-sidebar-accent hover:text-sidebar-accent-foreground focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 aria-disabled:pointer-events-none aria-disabled:opacity-50 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 [&>svg]:text-sidebar-accent-foreground data-[active=true]:bg-sidebar-accent data-[active=true]:text-sidebar-accent-foreground text-sm group-data-[collapsible=icon]:hidden"
                      data-sidebar="menu-sub-button"
                      data-size="md"
                      href="https://rail-signalling-info.netlify.app/"
                    >
                      Rail Signalling Info
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="absolute bottom-0 min-w-full flex p-2.5 border-t-2 border-t-[rgba(0,0,0,0.1)]"
      >
        <a
          class="px-2 py-1 rounded-md text-sm flex gap-2 items-center hover:bg-[hsl(240,4.8%,95.9%)] min-w-full"
          href="/about"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 18a4 4 0 0 0-8 0"
            />
            <circle
              cx="12"
              cy="11"
              r="3"
            />
            <rect
              height="18"
              rx="2"
              width="18"
              x="3"
              y="4"
            />
            <line
              x1="8"
              x2="8"
              y1="2"
              y2="4"
            />
            <line
              x1="16"
              x2="16"
              y1="2"
              y2="4"
            />
          </svg>
          About
        </a>
      </div>
    </div>
  </div>
</div>
`;
